### PR TITLE
enhance deployment for ubuntu

### PR DESCRIPTION
**Pull request by dalanlan**
- Fix the legacy problem which build.sh works heavily based on current dir. See also #13900 
- Download binaries for kube-up automatically

/cc @mikedanese

### CODE REVIEW

File: cluster/ubuntu/build.sh [hunk 1]
```diff
@@ -26,12 +26,15 @@ function cleanup {
 }
 trap cleanup SIGHUP SIGINT SIGTERM
 
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+pushd ${KUBE_ROOT}/cluster/ubuntu
```
Comment (mikedanese) [inline cluster/ubuntu/build.sh]: still depends on a working directory?

File: cluster/ubuntu/util.sh [hunk 2]
```diff
@@ -446,15 +441,19 @@ function prepare-push() {
     fi
   else
     # Run build.sh to get the required release 
-    pushd ubuntu
-    source "build.sh"    
-    popd
+    KUBE_VERSION=${KUBE_VERSION} "${KUBE_ROOT}/cluster/ubuntu/build.sh"
```
Comment (mikedanese) [inline cluster/ubuntu/util.sh]: I think you could reasonbly just export KUBE_VERSION.

File: cluster/ubuntu/build.sh (hunk 1 above)
Comment (mikedanese) [inline cluster/ubuntu/build.sh]: This is still a great imporvement since i can call ./cluster/ubuntu/build.sh from anywhere now.

**Comment by mikedanese** [comment]
Thanks @dalanlan! small comments

File: cluster/ubuntu/build.sh (hunk 1 above)
Comment (dalanlan) [inline cluster/ubuntu/build.sh]: I could be misreading, is there any problem with this? 
`pushd` here manages to make sure that the binaries dir locates expectedly.

**Comment by dalanlan** [comment]
Updated. @mikedanese

**Pull request merged**